preprocess/preprocess.go

```diff
@@ -96,11 +99,85 @@ func ExpandJSON(block *gabs.Container, vm *goja.Runtime) {
 	log.Debug().Str("Processed JSON block", block.String()).Msg("")
 }
 
-func LoadElfEnv(l2path string) {
+func SearchL2ConfigEnv(dir string) (string, error) {
+	parentDir := filepath.Dir(dir)
+	for parentDir != string(filepath.Separator) {
+		l2ConfigPath := filepath.Join(parentDir, "l2config.env")
+		_, err := os.Stat(l2ConfigPath)
+		if err == nil {
+			return l2ConfigPath, nil // Found the l2config.env file
+		}
+		parentDir = filepath.Dir(parentDir)
+	}
+	return "", errors.New("Didn't find l2config.env in the API directory")
+}
+
+func LoadEnvFile(l2path string) {
+	envFileName := filepath.Base(l2path)
 	err := godotenv.Load(l2path)
 	if err != nil {
-		log.Info().Str("Type", "Preprocess").Msg("Didn't find l2.env in the API directory")
+		log.Info().Str("Type", "Preprocess").Msg("Didn't find "+ envFileName +" in the API directory")
+	}
+}
+
+func LoadEnvironments(dir string) {
+	l2ConfigPath, err := SearchL2ConfigEnv(dir)
+	if err != nil {
+		log.Info().Str("Type", "Preprocess").Msg(err.Error())
+	} else {
+		LoadEnvFile(l2ConfigPath) // Loads global variables from l2config.env
+	}
+	LoadEnvFile(path.Join(dir, "l2.env")) // Overwrites the global variables if declared again in l2.env
+}
+
+func readFile(filename string) (envMap map[string]string, err error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return
+	}
+	defer file.Close()
+	return godotenv.Parse(file)
```

Comment:
One thing to verify is - does this properly handle back-tick commands?

Say I have an echo statement or something, is it going to do the right thing?

Reply:
Like this in the l2.env file ?
```
export ECHO_CMD="`echo "base64 image.jpeg"`"

```

Reply:
export PHOTO=`base64 image.jpeg`
?

Reply:
yes, the photo case must work (check tests/examples). 

in general, the backticks can have any arbitrary linux command - they run first, the output is stored in the env variable ultimately.

Reply:
No there is no test on that, will add